gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
+                                           (self.dsi['lon'] < -90 + lon_resolution/2),
+                                           drop=True).mean('lon')
+
+        # calculate latitude angle
+        lat_step = 180 / lat_points
+
+        # avarage over latitude space
+        spectra_list = []
+        for lat in range(lat_points):
+            tmp_morning = ds_clouds_morning.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+            tmp_evening = ds_clouds_evening.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+
+            # get temperature profile
+            temp = tmp_morning['T'].values[::-1]
+            # calculate chemistry in mass fractions
+            if mass_frac is None:
+                abus = interpol_abundances(co_ratios, feh_ratios, temp, pres)
+                for key in tlist:
+                    if key[0] in abus:
+                        abus[key[1]] = abus[key[0]]
+            else:
+                abus = mass_frac[0][lat]
+            # calcualte transmision spectra for the morning terminator of current lat point
+            self.prt.calc_transm(temp,
+                                 abus,
+                                 gravity,
+                                 mmw*np.ones_like(temp),
+                                 R_pl=rplanet*100,
+                                 P0_bar=pressure_0)
+            # add the spectra to the list
+            spectra_list.append(self.prt.transm_rad)
+
+            # get temperature profile
+            temp = tmp_evening['T'].values[::-1]
+            # calculate chemistry in mass fractions
+            if mass_frac is None:
+                abus = interpol_abundances(co_ratios, feh_ratios, temp, pres)
+                for key in tlist:
+                    if key[0] in abus:
+                        abus[key[1]] = abus[key[0]]
+            else:
+                abus = mass_frac[1][lat]
+            # calcualte transmision spectra for the morning terminator of current lat point
+            self.prt.calc_transm(temp,
+                                 abus,
+                                 gravity,
+                                 mmw*np.ones_like(temp),
+                                 R_pl=rplanet*100,
+                                 P0_bar=pressure_0)
+            # add the spectra to the list
+            spectra_list.append(self.prt.transm_rad)
+
+        # avarage over all profiles (area avaraged)
+        spectra = np.zeros_like(spectra_list[0])
+        i = 0
+        for profs in spectra_list:
+            spectra += profs**2
+            i += 1
+        spectra = np.sqrt(spectra/i)
```

Comment:
maybe use something like
```py
np.sqrt(np.sum(np.asarray(spectra_list)**2,axis=0))
```
instead?

Reply:
the division is missing, but other than that it is a nice short hand. I added it.